src/client/datascience/gather/gather.ts

```
@@ -26,35 +25,56 @@ export class GatherExecution implements IGatherExecution {
         @inject(IDisposableRegistry) private disposables: IDisposableRegistry,
         @inject(ICommandManager) private commandManager: ICommandManager
     ) {
-        this._enabled = this.configService.getSettings().datascience.enableGather ? true : false;
-
-        this.dataflowAnalyzer = new DataflowAnalyzer();
-        this._executionSlicer = new ExecutionLogSlicer(this.dataflowAnalyzer);
+        this._enabled =
+            this.configService.getSettings().datascience.enableGather &&
+            this.configService.getSettings().insidersChannel !== 'off'
+                ? true
+                : false;
 
         if (this._enabled) {
-            this.disposables.push(
-                this.configService.getSettings(undefined).onDidChange(e => this.updateEnableGather(e))
-            );
+            try {
+                // tslint:disable-next-line: no-require-imports
+                const ppa = require('@msrvida/python-program-analysis') as typeof import('@msrvida-python-program-analysis');
+
+                this.dataflowAnalyzer = new ppa.DataflowAnalyzer();
+                this._executionSlicer = new ppa.ExecutionLogSlicer(this.dataflowAnalyzer);
+
+                if (this._enabled) {
+                    this.disposables.push(
+                        this.configService.getSettings(undefined).onDidChange(e => this.updateEnableGather(e))
+                    );
+                }
+                traceInfo('Gathering tools have been activated');
+            } catch (ex) {
+                traceInfo('Gathering tools could not be activated');
```

Why would we want such a try..catch.
If it is enabled, i would expect it to fail early. Or we should display a message about this not working and revert `this._enabled`.
Else its possible something goes wrong and we'll never know about this